### PR TITLE
19 create incident user story

### PR DESCRIPTION
**Pull request by garrett-vangilder**
## Status
**READY**

## Testing

[ ] I have added passing tests for all new functionality.

[X] I have confirmed that all test prior to this pull request still PASS.

## Documentation

[ ] I have fully documented features in this PR's code. Please put an `X` in the box to confirm.

[X ] This PR does **NOT** require documentation.

## Description
A few sentences describing the overall goals of the pull request's commits.

This partially resolves issue number 19. I think that this issue needs to be broken down into smaller issues. So, this one partially resolves the issue at hand.

## Resolves Issue Number

#19 

## Deploy Notes
Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.

You will need to add these methods into main on program

```
            LoginAction.ReadInput();
            CreateIncidentAction.ReadInput();
```

```sh
git pull --prune
git checkout 19-create-incident-user-story
dotnet run
```

## Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.

1. Run the program and add 2 different incidents.
2. You can see that this program works, because the numbers increment properly 

## Impacted Files in Application
List general components of the application that this PR will affect:

* actions/CreateIncidentAction.cs

### CODE REVIEW

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (matthamil) [inline src/customer-service/Actions/CreateIncidentAction.cs]: App crashes when a resolution is added and the incident is saved.

**Review by matthamil — CHANGES_REQUESTED** [review]
See comments

**Comment by garrett-vangilder** [comment]
I'd love to hold off on this till I make some changes.

**Comment by garrett-vangilder** [comment]
@TeamCharles/team This PR is ready for your review!

**Review by matthamil — APPROVED** [review]

**Comment by matthamil** [comment]
LGTM 🎸

File: src/customer-service/Factories/CustomerFactory.cs
Comment (daynewright) [inline src/customer-service/Factories/CustomerFactory.cs:127]: This fails if no spaces are entered when submitting the customer name.

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (daynewright) [inline src/customer-service/Actions/CreateIncidentAction.cs]: We need to check if the customer as entered exists before assigning the id.  If it is typed wrong then this fails.

Comment (daynewright) [reply src/customer-service/Actions/CreateIncidentAction.cs]: I think you could do a `try catch`  and fix this?

**Review by daynewright — CHANGES_REQUESTED** [review]

File: customer-service-tests/CustomerFactoryTests.cs
Comment (MattKraatz) [inline customer-service-tests/CustomerFactoryTests.cs:71]: maybe also assert that the count is greater than 1:
`Assert.True(customers.Count() > 1)` or something like that

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (MattKraatz) [inline src/customer-service/Actions/CreateIncidentAction.cs]: We need the `LabelFactory` here too

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (MattKraatz) [inline src/customer-service/Actions/CreateIncidentAction.cs]: It would be more readable to instantiate these factories in the context in which they are used

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (MattKraatz) [inline src/customer-service/Actions/CreateIncidentAction.cs]: What is this used for?

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (MattKraatz) [reply src/customer-service/Actions/CreateIncidentAction.cs]: You can use linq to query your `customerList` enumerable for a matching FirstName and LastName. Load that into a while loop. No need to hit the database again

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (MattKraatz) [inline src/customer-service/Actions/CreateIncidentAction.cs]: this breaks if the user enters a non-integer

Comment (MattKraatz) [reply src/customer-service/Actions/CreateIncidentAction.cs]: could use a Try/Catch block here

Comment (MattKraatz) [reply src/customer-service/Actions/CreateIncidentAction.cs]: basically every time we're waiting for input, that needs to be in a while loop

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (MattKraatz) [inline src/customer-service/Actions/CreateIncidentAction.cs]: this breaks if the user enters a non-integer

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (MattKraatz) [reply src/customer-service/Actions/CreateIncidentAction.cs]: I don't experience this issue. Must have been resolved with changes.

**Review by Anulfo — COMMENTED** [review]
LGTM

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (MattKraatz) [inline src/customer-service/Actions/CreateIncidentAction.cs]: Need to attach the EmployeeId to the incident.

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (MattKraatz) [inline src/customer-service/Actions/CreateIncidentAction.cs]: Need to attach the IncidentTypeId to the Incident

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (MattKraatz) [inline src/customer-service/Actions/CreateIncidentAction.cs]: this should be the count + 1, like an auto-incremented value. Basically when I add Incident #5, this WriteLine tells me I added Incident #4

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (MattKraatz) [inline src/customer-service/Actions/CreateIncidentAction.cs]: Need to print a list of all applicable labels here. Check out Dayne's Single Incident PR

Comment (MattKraatz) [reply src/customer-service/Actions/CreateIncidentAction.cs]: This screen should look exactly like Dayne's, actually. Might help to re-review the requirements in case we're off-base on some other mockups.

**Review by MattKraatz — CHANGES_REQUESTED** [review]
Getting there...

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (garrett-vangilder) [reply src/customer-service/Actions/CreateIncidentAction.cs]: Changed in previous PR

File: customer-service-tests/CustomerFactoryTests.cs
Comment (garrett-vangilder) [reply customer-service-tests/CustomerFactoryTests.cs:71]: Added

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (garrett-vangilder) [reply src/customer-service/Actions/CreateIncidentAction.cs]: I will change this for the good of the team! 👥

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (garrett-vangilder) [reply src/customer-service/Actions/CreateIncidentAction.cs]: Added

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (garrett-vangilder) [reply src/customer-service/Actions/CreateIncidentAction.cs]: Removed

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (garrett-vangilder) [reply src/customer-service/Actions/CreateIncidentAction.cs]: Added

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (garrett-vangilder) [reply src/customer-service/Actions/CreateIncidentAction.cs]: added

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (garrett-vangilder) [reply src/customer-service/Actions/CreateIncidentAction.cs]: Word, my mistake I thought that the `Count` method was not zero based

this has been added

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (MattKraatz) [inline src/customer-service/Actions/CreateIncidentAction.cs]: How about we leave off `CustomerId` from this WriteLine to eliminate confusion for the user

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (MattKraatz) [inline src/customer-service/Actions/CreateIncidentAction.cs:31]: Let's put this after the list of customers

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (MattKraatz) [inline src/customer-service/Actions/CreateIncidentAction.cs]: We need a `Console.ReadLine()` here to start, else the user is shown the "This is not a valid answer. Try again!" right away

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (MattKraatz) [inline src/customer-service/Actions/CreateIncidentAction.cs]: The program never enters this `while` loop as far as I can tell. I think you want: `orderID == 0`

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (MattKraatz) [inline src/customer-service/Actions/CreateIncidentAction.cs]: Lose the code below here. Instead, set the `IncidentFactory.Instance.ActiveInstance` to the current instance and the `CustomerFactory.Instance.ActiveCustomer` to the current customer. Then call Dayne's `ShowSingleIncidentAction.ReadLine()`.

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (MattKraatz) [inline src/customer-service/Actions/CreateIncidentAction.cs]: NVM

**Review by MattKraatz — CHANGES_REQUESTED** [review]
A few final changes and we should be good to merge 👍

**Review by daynewright — APPROVED** [review]

**Comment by daynewright** [comment]
LGTM after @MattKraatz  changes.

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (garrett-vangilder) [reply src/customer-service/Actions/CreateIncidentAction.cs:31]: added

File: src/customer-service/Actions/CreateIncidentAction.cs
Comment (garrett-vangilder) [reply src/customer-service/Actions/CreateIncidentAction.cs]: changed

**Review by MattKraatz — APPROVED** [review]

**Pull request merged**
